File: App.Client/src/api/services/Admin/role/models/QueryRolePagedOutput.ts
import { RoleTypeEnum } from "@/api/enums/RoleTypeEnum";
import { DataScopeTypeEnum } from "@/api/enums/DataScopeTypeEnum";

/**
 * Fast.Admin.Service.Role.Dto.QueryRolePagedOutput 获取角色分页列表输出
 */
export interface QueryRolePagedOutput {
  /**
   * 角色Id
   */
  roleId?: number;
  /**
   * 
   */
  roleType?: RoleTypeEnum;
  /**
   * 角色名称
   */
  roleName?: string;
  /**
   * 角色编码
   */
  roleCode?: string;
  /**
   * 排序
   */
  sort?: number;
  /**
   * 
   */
  dataScopeType?: DataScopeTypeEnum;
  /**
   * 备注
   */
  remark?: string;
  /**
   * 
   */
  departmentName?: string;
  /**
   * 
   */
  createdUserName?: string;
  /**
   * 
   */
  createdTime?: Date;
  /**
   * 
   */
  updatedUserName?: string;
  /**
   * 
   */
  updatedTime?: Date;
  /**
   * 
   */
  rowVersion?: number;
}

File: Api.Server/src/Admin.Service/Role/Dto/QueryRolePagedOutput.cs
// Copyright © 2018-Present 小方
// SPDX-License-Identifier: Apache-2.0
// 
// 本文件依据 Apache License 2.0 授权，完整条款见仓库根目录 LICENSE。
// 本软件按“原样”提供，相关免责声明及责任限制以许可证及适用法律为准。
// 版权来源、合法使用与二次开发责任说明见仓库根目录 README.md。

namespace Fast.Admin.Service.Role.Dto;

/// <summary>
/// 获取角色分页列表输出
/// </summary>
public class QueryRolePagedOutput : PagedOutput
{
    /// <summary>
    /// 角色Id
    /// </summary>
    public long RoleId { get; set; }

    /// <summary>
    /// 角色类型
    /// </summary>
    public RoleTypeEnum RoleType { get; set; }

    /// <summary>
    /// 是否使用系统菜单
    /// </summary>
    public bool IsSystemMenu { get; set; }

    /// <summary>
    /// 角色名称
    /// </summary>
    [SugarSearchValue]
    public string RoleName { get; set; }

    /// <summary>
    /// 角色编码
    /// </summary>
    [SugarSearchValue]
    public string RoleCode { get; set; }

    /// <summary>
    /// 排序
    /// </summary>
    public int Sort { get; set; }

    /// <summary>
    /// 数据范围类型
    /// </summary>
    public DataScopeTypeEnum DataScopeType { get; set; }

    /// <summary>
    /// 备注
    /// </summary>
    public string Remark { get; set; }
}
